epoxy-processor/src/main/java/com/airbnb/epoxy/processor/Memoizer.kt

```diff
@@ -355,6 +355,16 @@ class Memoizer(
             }
         }
     }
+
+    private val implementsModelCollectorMap = mutableMapOf<GeneratedModelInfo, Boolean>()
+    fun implementsModelCollector(modelInfo: GeneratedModelInfo): Boolean {
+        return synchronized(typeMap) {
+            val implementsModelCollector = modelInfo.superClassElement.interfaces.firstOrNull {
+                it.toString() == ClassNames.MODEL_COLLECTOR.toString()
+            } != null
+            implementsModelCollectorMap.getOrPut(modelInfo) { implementsModelCollector }
```

Comment:
`implementsModelCollector` is always calculated, even when the map already contains the value.

Could you remove the `implementsModelCollector` intermediate property and just put the whole calculation in the `getOrPut` lambda?